parquet-hadoop/src/main/java/org/apache/parquet/hadoop/util/HadoopStreams.java

```diff
@@ -66,6 +67,15 @@ public static SeekableInputStream wrap(FSDataInputStream stream) {
     }
   }
 
+  private static boolean isWrappedStreamByteBufferReadable(FSDataInputStream stream) {
+    InputStream wrapped = stream.getWrappedStream();
+    if (wrapped instanceof FSDataInputStream) {
+      return isWrappedStreamByteBufferReadable(((FSDataInputStream) wrapped));
```

Comment:
Is there a corner case that can cause an infinite loop?

Reply:
Yes, it could be. But it may be hard to create such a case. As its code shows, FSDataInputStream is a wrapper class of an inputstream. When we check the wrapped inputstream recursively, it would finally reach an inputstream whose type is not FSDataInputStream. A developer could override `getWrappedStream` as `return this`, while this makes no sence.

Reply:
I understand it would be very rare case but once that happen it would be hard to debug this 'hang' issue. Let's do two things: 1) Add check if it is 'this'; Throw exception if that happens; 2) Add debug log; When it hangs, developer can enable debug log and see what parquet-mr is doing.

Reply:
Good suggestions! I have updated as the comment.

Reply:
you can't. the inner stream is set in the constructor, so cannot take the not-yet-constructed class as an argument.